go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -565,6 +565,408 @@ var DoltMerge = []enginetest.ScriptTest{
+		Name: "column drop and recreate with different type that can be coerced (int -> string)",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c int);",
+			"insert into t values (1, 2), (3, 4);",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"alter table t drop column c;",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'dropping column c'));",
+
+			"alter table t add column c text;",
+			"insert into t values (100, '101');",
+			"set @Commit3 = (select DOLT_COMMIT('-am', 're-adding column c'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
+				Expected: []sql.Row{{5}},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit1 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, "2", nil, nil, "added"},
+					{3, "4", nil, nil, "added"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, nil, 1, "2", "modified"},
+					{3, nil, 3, "4", "modified"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit3 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{100, "101", nil, nil, "added"},
+				},
+			},
+		},
+	},
+	{
+		Name: "column drop and recreate with different type that can NOT be coerced (string -> int)",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c text);",
+			"insert into t values (1, 'two'), (3, 'four');",
+			"set @Commit1 = (select DOLT_COMMIT('-am', 'creating table t'));",
+
+			"alter table t drop column c;",
+			"set @Commit2 = (select DOLT_COMMIT('-am', 'dropping column c'));",
+
+			"alter table t add column c int;",
+			"insert into t values (100, 101);",
+			"set @Commit3 = (select DOLT_COMMIT('-am', 're-adding column c'));",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM DOLT_DIFF_t;",
+				Expected: []sql.Row{{5}},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit1 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, nil, nil, nil, "added"},
+					{3, nil, nil, nil, "added"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit2 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{1, nil, 1, nil, "modified"},
+					{3, nil, 3, nil, "modified"},
+				},
+			},
+			{
+				Query: "SELECT to_pk, to_c, from_pk, from_c, diff_type FROM DOLT_DIFF_t WHERE TO_COMMIT=@Commit3 ORDER BY to_pk;",
+				Expected: []sql.Row{
+					{100, 101, nil, nil, "added"},
+				},
+			},
+			{
+				// TODO: What's the best way to test show warnings? With a custom/non-scripted test?
```

Comment:
You can clear warnings in between different test cases so that you know which warnings came from which statements.

Reply:
I extended the ScriptTest test utils in go-mysql-server so that we can express more expectations about warnings: 
https://github.com/dolthub/go-mysql-server/pull/810

Once that GMS PR is merged in and available to Dolt, I'll push the update for these tests to use those new members.